src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReconcileDatasetPidCommand.java

```diff
@@ -0,0 +1,181 @@
+        if (oldId == null) {
+            throw new IllegalStateException("Dataset without a global identifier, cannot alter!");
+        }
+        logger.fine("Reconciling dataset( id =`" + getDataset().getId() + ")` - removing globalId `" + getDataset().getGlobalId() + '`');
+        // remove dataset PID
+        try {
+            if (currentPidProvider.alreadyRegistered(getDataset())) { //if not registered with PIDProvider than there is no need to delete it...
+                currentPidProvider.deleteIdentifier(getDataset()); // delete it externally
+            }
+            getDataset().setGlobalId(null); //  remove it internally
+            getDataset().setGlobalIdCreateTime(null);
+            getDataset().setIdentifierRegistered(false);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "Identifier deletion was not successful:", e.getMessage());
+        }
+
+        if(ctxt.systemConfig().isFilePIDsEnabledForCollection(getDataset().getOwner())) {
+            reconcileFilePids(ctxt, currentPidProvider);
+        }
+        getDataset().setPidGenerator(this.newPidProvider);
+        newPidProvider.generatePid(getDataset()); // this updates Protocol, Authority, and Identifier and thus a new GlobalID
+        logger.fine("Reconciling dataset( id =`" + getDataset().getId() + ")` - creating new globalId `" + getDataset().getGlobalId() + '`');
+        if (!newPidProvider.registerWhenPublished()) {
+            registerExternalIdentifier(getDataset(), ctxt, true); // this updates GlobalIdCreateTime and IdentifierRegistered
+        }
+        // keep old Id als alternative identifier
+        AlternativePersistentIdentifier api;
+        api = new AlternativePersistentIdentifier();
+        api.setProtocol(oldId.getProtocol());
+        api.setAuthority(oldId.getAuthority());
+        api.setIdentifier(oldId.getIdentifier());
+        api.setDvObject(getDataset());
+        api.setStorageLocationDesignator(true);// cf. Dataset#getIdentifierForFileStorage()
+        if(getDataset().getAlternativePersistentIndentifiers()!=null) {
+            getDataset().getAlternativePersistentIndentifiers().add(api);
+        }else{
+            getDataset().setAlternativePersistentIndentifiers(Set.of(api));
+        }
+        // We keep the old persistent identifier as AlternativePersistentIdentifier with storageLocationDesignator true.
+        // This keep the link the object store intact, without altering the files.
+        // IMHO: This command should also update the storage. First, maintenance of the storage becomes a mess if there is a counterintuitive layout.
+        // Second, it could occur that another object is minted with the old identifier and consequently we have a conflict in our storage system.
+        // We accept this risk for now since a super user can update the storage manually from old file path to the new one, and remove the AlternativeIdentifer form the database.
```

Comment:
FWIW: If changes above don't remove this risk (by making sure you still find the alternateId when checking to see if a newly generated one already exists), then a manual work-around would be to add the alternateID to the prohibited list for the PID provider.

Reply:
Right, the current solution is the addition of the old id as an alternativeIdentifier and the integration of alternativeIdentifers in the `isGlobalIdLocallyUnique` check of the PIDproivder.